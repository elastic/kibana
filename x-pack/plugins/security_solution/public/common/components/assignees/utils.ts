/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { NO_ASSIGNEES_VALUE } from './constants';
import type { AssigneesIdsSelection } from './types';

export const removeNoAssigneesSelection = (assignees: AssigneesIdsSelection[]): string[] =>
  assignees.filter<string>((assignee): assignee is string => assignee !== NO_ASSIGNEES_VALUE);
