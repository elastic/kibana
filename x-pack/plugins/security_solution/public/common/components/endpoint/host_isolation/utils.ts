/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import type { EndpointPendingActions } from '../../../../../common/endpoint/types';
import type { PendingActions } from './endpoint_host_isolation_status';

export const getPendingActions = (
  pendingActions: EndpointPendingActions['pending_actions'] = {}
): PendingActions => ({
  pendingIsolate: pendingActions?.isolate ?? 0,
  pendingUnIsolate: pendingActions?.unisolate ?? 0,
  pendingKillProcess: pendingActions?.['kill-process'] ?? 0,
  pendingSuspendProcess: pendingActions?.['suspend-process'] ?? 0,
  pendingRunningProcesses: pendingActions?.['running-processes'] ?? 0,
  pendingGetFile: pendingActions?.['get-file'] ?? 0,
  pendingExecute: pendingActions?.execute ?? 0,
});
