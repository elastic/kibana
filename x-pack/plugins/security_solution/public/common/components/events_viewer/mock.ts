/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export const mockEventViewerResponse = {
  totalCount: 12,
  pageInfo: {
    activePage: 0,
    fakeTotalCount: 100,
  },
  events: [],
};
