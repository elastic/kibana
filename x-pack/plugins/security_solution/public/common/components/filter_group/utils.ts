/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import type {
  ControlGroupInput,
  ControlPanelState,
  OptionsListEmbeddableInput,
} from '@kbn/controls-plugin/common';

export const getPanelsInOrderFromControlsInput = (controlInput: ControlGroupInput) => {
  const panels = controlInput.panels;

  return Object.values(panels).sort((a, b) => a.order - b.order);
};

export const getFilterItemObjListFromControlInput = (controlInput: ControlGroupInput) => {
  const panels = getPanelsInOrderFromControlsInput(controlInput);
  return panels.map((panel) => {
    const {
      explicitInput: { fieldName, selectedOptions, title, existsSelected, exclude },
    } = panel as ControlPanelState<OptionsListEmbeddableInput>;

    return {
      fieldName: fieldName as string,
      selectedOptions: selectedOptions ?? [],
      title,
      existsSelected: existsSelected ?? false,
      exclude: exclude ?? false,
    };
  });
};
