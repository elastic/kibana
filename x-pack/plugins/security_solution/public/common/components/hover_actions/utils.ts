/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { BrowserField } from '../../containers/source';

export const getAdditionalScreenReaderOnlyContext = ({
  field,
  value,
}: {
  field: string;
  value?: string[] | string | null;
}): string => {
  if (value == null) {
    return field;
  }

  return Array.isArray(value) ? `${field} ${value.join(' ')}` : `${field} ${value}`;
};

export const allowTopN = ({
  browserField,
  fieldName,
}: {
  browserField: Partial<BrowserField> | undefined;
  fieldName: string;
}): boolean => {
  const isAggregatable = browserField?.aggregatable ?? false;
  const fieldType = browserField?.type ?? '';
  const isAllowedType = [
    'boolean',
    'geo-point',
    'geo-shape',
    'ip',
    'keyword',
    'number',
    'numeric',
    'string',
  ].includes(fieldType);

  // TODO: remove this explicit allowlist when the ECS documentation includes alerts
  const isAllowlistedNonBrowserField = [
    'signal.ancestors.depth',
    'signal.ancestors.id',
    'signal.ancestors.rule',
    'signal.ancestors.type',
    'signal.original_event.action',
    'signal.original_event.category',
    'signal.original_event.code',
    'signal.original_event.created',
    'signal.original_event.dataset',
    'signal.original_event.duration',
    'signal.original_event.end',
    'signal.original_event.hash',
    'signal.original_event.id',
    'signal.original_event.kind',
    'signal.original_event.module',
    'signal.original_event.original',
    'signal.original_event.outcome',
    'signal.original_event.provider',
    'signal.original_event.risk_score',
    'signal.original_event.risk_score_norm',
    'signal.original_event.sequence',
    'signal.original_event.severity',
    'signal.original_event.start',
    'signal.original_event.timezone',
    'signal.original_event.type',
    'signal.original_time',
    'signal.parent.depth',
    'signal.parent.id',
    'signal.parent.index',
    'signal.parent.rule',
    'signal.parent.type',
    'signal.rule.created_by',
    'signal.rule.description',
    'signal.rule.enabled',
    'signal.rule.false_positives',
    'signal.rule.filters',
    'signal.rule.from',
    'signal.rule.id',
    'signal.rule.immutable',
    'signal.rule.index',
    'signal.rule.interval',
    'signal.rule.language',
    'signal.rule.max_signals',
    'signal.rule.name',
    'signal.rule.note',
    'signal.rule.output_index',
    'signal.rule.query',
    'signal.rule.references',
    'signal.rule.risk_score',
    'signal.rule.rule_id',
    'signal.rule.saved_id',
    'signal.rule.severity',
    'signal.rule.size',
    'signal.rule.tags',
    'signal.rule.threat',
    'signal.rule.threat.tactic.id',
    'signal.rule.threat.tactic.name',
    'signal.rule.threat.tactic.reference',
    'signal.rule.threat.technique.id',
    'signal.rule.threat.technique.name',
    'signal.rule.threat.technique.reference',
    'signal.rule.timeline_id',
    'signal.rule.timeline_title',
    'signal.rule.to',
    'signal.rule.type',
    'signal.rule.updated_by',
    'signal.rule.version',
    'signal.status',
  ].includes(fieldName);

  return isAllowlistedNonBrowserField || (isAggregatable && isAllowedType);
};
