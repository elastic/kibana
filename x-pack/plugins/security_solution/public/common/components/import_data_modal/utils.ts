/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import type { ErrorToastOptions, Toast, ToastInput, ToastOptions } from '@kbn/core/public';

import * as i18n from './translations';

import type {
  ImportDataResponse,
  ImportResponseError,
  ImportRulesResponseError,
  ExceptionsImportError,
} from '../../../detection_engine/rule_management/logic';

export const formatError = (
  i18nFailedDetailedMessage: (message: string) => string,
  importResponse: ImportDataResponse,
  errors: Array<ImportRulesResponseError | ImportResponseError | ExceptionsImportError>
) => {
  const formattedErrors = errors.map((e) => i18nFailedDetailedMessage(e.error.message));
  const error: Error & { raw_network_error?: object } = new Error(formattedErrors.join('. '));
  error.stack = undefined;
  error.name = 'Network errors';
  error.raw_network_error = importResponse;

  return error;
};

const mapErrorMessageToUserMessage = (
  actionConnectorsErrors: Array<ImportRulesResponseError | ImportResponseError>
) => {
  return actionConnectorsErrors.map((connectorError) => {
    const { error } = connectorError;
    const { status_code: statusCode, message: originalMessage } = error || {};
    let message;
    switch (statusCode) {
      case 403:
        message = i18n.ACTION_CONNECTORS_ADDITIONAL_PRIVILEGES;
        break;

      default:
        message = originalMessage;

        break;
    }

    return { ...connectorError, error: { ...error, message } };
  });
};

export const showToasterMessage = ({
  importResponse,
  exceptionsIncluded,
  actionConnectorsIncluded,
  successMessage,
  errorMessage,
  errorMessageDetailed,
  addError,
  addSuccess,
}: {
  importResponse: ImportDataResponse;
  exceptionsIncluded: boolean;
  actionConnectorsIncluded: boolean;
  successMessage: (totalCount: number) => string;
  errorMessage: (totalCount: number) => string;
  errorMessageDetailed: (message: string) => string;
  addError: (error: unknown, options: ErrorToastOptions) => Toast;
  addSuccess: (toastOrTitle: ToastInput, options?: ToastOptions | undefined) => Toast;
}) => {
  if (importResponse.success) {
    if (importResponse.success_count > 0) addSuccess(successMessage(importResponse.success_count));
    if (
      exceptionsIncluded &&
      importResponse.exceptions_success &&
      importResponse.exceptions_success_count != null &&
      importResponse.exceptions_success_count > 0
    ) {
      addSuccess(i18n.SUCCESSFULLY_IMPORTED_EXCEPTIONS(importResponse.exceptions_success_count));
    }
    if (
      actionConnectorsIncluded &&
      importResponse.action_connectors_success &&
      importResponse.action_connectors_success_count != null &&
      importResponse.action_connectors_success_count > 0
    ) {
      addSuccess(
        i18n.SUCCESSFULLY_IMPORTED_CONNECTORS(importResponse.action_connectors_success_count)
      );
    }
    return;
  }

  if (importResponse.errors.length > 0) {
    if (
      actionConnectorsIncluded &&
      importResponse.action_connectors_errors != null &&
      importResponse.action_connectors_errors.length > 0
    ) {
      const userErrorMessages = mapErrorMessageToUserMessage(
        importResponse.action_connectors_errors
      );
      const connectorError = formatError(errorMessageDetailed, importResponse, userErrorMessages);

      return addError(connectorError, {
        title: i18n.IMPORT_CONNECTORS_FAILED(userErrorMessages.length),
      });
    }
    const ruleError = formatError(errorMessageDetailed, importResponse, importResponse.errors);
    addError(ruleError, { title: errorMessage(importResponse.errors.length) });

    if (
      exceptionsIncluded &&
      importResponse.exceptions_errors != null &&
      importResponse.exceptions_errors.length > 0
    ) {
      const exceptionError = formatError(
        errorMessageDetailed,
        importResponse,
        importResponse.exceptions_errors
      );

      addError(exceptionError, {
        title: i18n.IMPORT_FAILED(importResponse.exceptions_errors.length),
      });
    }
  }
};
