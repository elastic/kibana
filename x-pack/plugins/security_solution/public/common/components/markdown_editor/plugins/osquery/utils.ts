/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { isEmpty, reduce } from 'lodash';

export const convertECSMappingToObject = (
  ecsMapping: Array<{
    key: string;
    result: {
      type: string;
      value: string;
    };
  }>
) =>
  reduce(
    ecsMapping,
    (acc, value) => {
      if (!isEmpty(value?.key) && !isEmpty(value.result?.type) && !isEmpty(value.result?.value)) {
        acc[value.key] = {
          [value.result.type]: value.result.value,
        };
      }

      return acc;
    },
    {} as Record<string, { field?: string; value?: string }>
  );
