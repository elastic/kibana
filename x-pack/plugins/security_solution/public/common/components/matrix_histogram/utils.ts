/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { ScaleType, Position } from '@elastic/charts';
import { get, groupBy, map, toPairs } from 'lodash/fp';

import { UpdateDateRange, ChartSeriesData } from '../charts/common';
import { MatrixHistogramMappingTypes, BarchartConfigs } from './types';
import { MatrixHistogramData } from '../../../../common/search_strategy';
import { histogramDateTimeFormatter } from '../utils';

interface GetBarchartConfigsProps {
  chartHeight?: number;
  from: string;
  legendPosition?: Position;
  to: string;
  onBrushEnd: UpdateDateRange;
  yTickFormatter?: (value: number) => string;
  showLegend?: boolean;
}

export const DEFAULT_CHART_HEIGHT = 174;
export const DEFAULT_Y_TICK_FORMATTER = (value: string | number): string => value.toLocaleString();

export const getBarchartConfigs = ({
  chartHeight,
  from,
  legendPosition,
  to,
  onBrushEnd,
  yTickFormatter,
  showLegend,
}: GetBarchartConfigsProps): BarchartConfigs => ({
  series: {
    xScaleType: ScaleType.Time,
    yScaleType: ScaleType.Linear,
    stackAccessors: ['g'],
  },
  axis: {
    xTickFormatter: histogramDateTimeFormatter([from, to]),
    yTickFormatter: yTickFormatter != null ? yTickFormatter : DEFAULT_Y_TICK_FORMATTER,
    tickSize: 8,
  },
  settings: {
    legendPosition: legendPosition ?? Position.Right,
    onBrushEnd,
    showLegend: showLegend ?? true,
    showLegendExtra: true,
    theme: {
      scales: {
        barsPadding: 0.08,
      },
      chartMargins: {
        left: 0,
        right: 0,
        top: 0,
        bottom: 0,
      },
      chartPaddings: {
        left: 0,
        right: 0,
        top: 0,
        bottom: 0,
      },
    },
  },
  customHeight: chartHeight ?? DEFAULT_CHART_HEIGHT,
});

export const defaultLegendColors = [
  '#1EA593',
  '#2B70F7',
  '#CE0060',
  '#38007E',
  '#FCA5D3',
  '#F37020',
  '#E49E29',
  '#B0916F',
  '#7B000B',
  '#34130C',
  '#GGGGGG',
];

export const formatToChartDataItem = ([key, value]: [
  string,
  MatrixHistogramData[]
]): ChartSeriesData => ({
  key,
  value,
});

export const getCustomChartData = (
  data: MatrixHistogramData[] | null,
  mapping?: MatrixHistogramMappingTypes
): ChartSeriesData[] => {
  if (!data) return [];
  const dataGroupedByEvent = groupBy('g', data);
  const dataGroupedEntries = toPairs(dataGroupedByEvent);
  const formattedChartData = map(formatToChartDataItem, dataGroupedEntries);
  return formattedChartData.map((item: ChartSeriesData, idx: number) => {
    const mapItem = get(item.key, mapping);
    return { ...item, color: mapItem?.color ?? defaultLegendColors[idx] };
  });
};
