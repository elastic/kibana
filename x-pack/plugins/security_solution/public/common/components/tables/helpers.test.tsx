/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import React from 'react';
import { shallow } from 'enzyme';

import '../../mock/match_media';
import {
  RowItemOverflowComponent,
  OverflowFieldComponent,
  getRowItemsWithActions,
} from './helpers';
import { TestProviders } from '../../mock';
import { useMountAppended } from '../../utils/use_mount_appended';
import { getEmptyValue } from '../empty_value';
import { render } from '@testing-library/react';

jest.mock('../../lib/kibana');

describe('Table Helpers', () => {
  const items = ['item1', 'item2', 'item3'];
  const mount = useMountAppended();

  describe('#getRowItemsWithActions', () => {
    test('it returns empty value when values is undefined', () => {
      const rowItem = getRowItemsWithActions({
        values: undefined,
        fieldName: 'attrName',
        idPrefix: 'idPrefix',
        aggregatable: false,
      });

      const { container } = render(<TestProviders>{rowItem}</TestProviders>);

      expect(container.textContent).toBe(getEmptyValue());
    });

    test('it returns empty string value when values is empty', () => {
      const rowItem = getRowItemsWithActions({
        values: [''],
        fieldName: 'attrName',
        idPrefix: 'idPrefix',
        aggregatable: false,
      });
      const { container } = render(<TestProviders>{rowItem}</TestProviders>);

      expect(container.textContent).toContain('(Empty string)');
    });

    test('it returns empty value when rowItem is null', () => {
      const rowItem = getRowItemsWithActions({
        values: null,
        fieldName: 'attrName',
        idPrefix: 'idPrefix',
        aggregatable: false,
        displayCount: 0,
      });
      const { container } = render(<TestProviders>{rowItem}</TestProviders>);
      expect(container.textContent).toBe(getEmptyValue());
    });

    test('it uses custom renderer', () => {
      const renderer = (item: string) => <>{`Hi ${item} renderer`}</>;
      const rowItem = getRowItemsWithActions({
        values: ['item1'],
        fieldName: 'attrName',
        idPrefix: 'idPrefix',
        aggregatable: false,
        render: renderer,
      });
      const { container } = render(<TestProviders>{rowItem}</TestProviders>);

      expect(container.textContent).toContain('Hi item1 renderer');
    });

    test('it returns no items when provided an empty array', () => {
      const rowItems = getRowItemsWithActions({
        values: [],
        fieldName: 'attrName',
        idPrefix: 'idPrefix',
        aggregatable: false,
      });
      const { container } = render(<TestProviders>{rowItems}</TestProviders>);
      expect(container.textContent).toBe(getEmptyValue());
    });

    test('it returns 2 items then overflows when displayCount is 2', () => {
      const rowItems = getRowItemsWithActions({
        values: items,
        fieldName: 'attrName',
        idPrefix: 'idPrefix',
        aggregatable: false,
        displayCount: 2,
      });
      const { queryAllByTestId, queryByTestId } = render(<TestProviders>{rowItems}</TestProviders>);

      expect(queryAllByTestId('cellActions-renderContent-attrName').length).toBe(2);
      expect(queryByTestId('overflow-button')).toBeInTheDocument();
    });
  });

  describe('#RowItemOverflow', () => {
    test('it returns correctly against snapshot', () => {
      const wrapper = shallow(
        <RowItemOverflowComponent
          values={items}
          fieldName="attrName"
          idPrefix="idPrefix"
          maxOverflowItems={1}
          overflowIndexStart={1}
          fieldType="keyword"
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    test('it does not show "more not shown" when maxOverflowItems are not exceeded', () => {
      const wrapper = shallow(
        <RowItemOverflowComponent
          values={items}
          fieldName="attrName"
          idPrefix="idPrefix"
          maxOverflowItems={5}
          overflowIndexStart={1}
          fieldType="keyword"
        />
      );
      expect(wrapper.find('[data-test-subj="popover-additional-overflow"]').length).toBe(0);
    });

    test('it shows correct number of overflow items when maxOverflowItems are not exceeded', () => {
      const wrapper = mount(
        <TestProviders>
          <RowItemOverflowComponent
            values={items}
            fieldName="attrName"
            idPrefix="idPrefix"
            maxOverflowItems={5}
            overflowIndexStart={1}
            fieldType="keyword"
          />
        </TestProviders>
      );
      wrapper.find('[data-test-subj="overflow-button"]').first().find('button').simulate('click');

      expect(
        wrapper.find('[data-test-subj="overflow-items"]').last().prop<JSX.Element[]>('children')
          ?.length
      ).toEqual(2);
    });

    test('it shows "more not shown" when maxOverflowItems are exceeded', () => {
      const wrapper = shallow(
        <RowItemOverflowComponent
          values={items}
          fieldName="attrName"
          idPrefix="idPrefix"
          maxOverflowItems={1}
          overflowIndexStart={1}
          fieldType="keyword"
        />
      );
      expect(wrapper.find('[data-test-subj="popover-additional-overflow"]').length).toBe(1);
    });
  });

  describe('OverflowField', () => {
    test('it returns correctly against snapshot', () => {
      const overflowString = 'This string is exactly fifty-one chars in length!!!';
      const wrapper = shallow(
        <OverflowFieldComponent value={overflowString} showToolTip={false} />
      );
      expect(wrapper).toMatchSnapshot();
    });

    test('it does not truncates as per custom overflowLength value', () => {
      const overflowString = 'This string is short';
      const wrapper = mount(<OverflowFieldComponent value={overflowString} overflowLength={20} />);
      expect(wrapper.text()).toBe('This string is short');
    });

    test('it truncates as per custom overflowLength value', () => {
      const overflowString = 'This string is exactly fifty-one chars in length!!!';
      const wrapper = mount(<OverflowFieldComponent value={overflowString} overflowLength={20} />);
      expect(wrapper.text()).toBe('This string is exact');
    });
  });
});
