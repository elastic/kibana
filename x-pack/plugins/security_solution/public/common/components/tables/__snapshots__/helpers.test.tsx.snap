// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Helpers #RowItemOverflow it returns correctly against snapshot 1`] = `
<Fragment>
  <Popover
    count={2}
    idPrefix="idPrefix"
  >
    <EuiText
      size="xs"
    >
      <MoreContainer
        fieldName="attrName"
        fieldType="keyword"
        idPrefix="idPrefix"
        moreMaxHeight="none"
        overflowIndexStart={1}
        values={
          Array [
            "item1",
            "item2",
            "item3",
          ]
        }
      />
      <p
        data-test-subj="popover-additional-overflow"
      >
        <EuiTextColor
          color="subdued"
        >
          1
           
          <FormattedMessage
            defaultMessage="more not shown"
            id="xpack.securitySolution.tables.rowItemHelper.moreDescription"
            values={Object {}}
          />
        </EuiTextColor>
      </p>
    </EuiText>
  </Popover>
</Fragment>
`;

exports[`Table Helpers OverflowField it returns correctly against snapshot 1`] = `
<span>
  This string is exactly fifty-one chars in length!!
  <EuiToolTip
    content="This string is exactly fifty-one chars in length!!!"
    delay="regular"
    display="inlineBlock"
    position="top"
  >
    <MoreRowItems
      type="boxesHorizontal"
    />
  </EuiToolTip>
</span>
`;
